src/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonValueConverterEnum.cs

```diff
@@ -39,20 +49,90 @@ public override bool TryRead(Type valueType, ref Utf8JsonReader reader, out TVal
                 return false;
             }
 
-            if (s_isUint64)
+            if (s_isInt32)
+            {
+                if (reader.TryGetInt32(out int int32))
```

Comment:
@jkotas would we be safe doing this?

``` C#
if (reader.TryGetInt32(out int int32))
{
    value = Unsafe.As<int, T>(ref int32);
    return true;
}
```

Reply:
This has endianness bug.

Reply:
Ah, ok. ReadUnaligned?

Reply:
> Ah, ok. ReadUnaligned?

Never mind- looking for the right call.

Reply:
Maybe I misunderstood - if you `Unsafe.As<int, T>` only when T is int32 or uint32, then this is certainly fine.

Reply:
> Maybe I misunderstood - if you Unsafe.As<int, T> only when T is int32 (or uint32), then it is certainly fine.

Ok :) I was getting a bit confused there. Yes, restricting to same size. Should avoid the boxing/unboxing.

Reply:
I'll update

Reply:
@JeremyKuhne @jkotas PTAL